partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelScanIndexed.kt

```diff
@@ -0,0 +1,39 @@
+package org.partiql.eval.internal.operator.rel
+
+import org.partiql.eval.internal.Record
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.value.CollectionValue
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.intValue
+import java.math.BigInteger
+
+@OptIn(PartiQLValueExperimental::class)
+internal class RelScanIndexed(
+    private val expr: Operator.Expr
+) : Operator.Relation {
+
+    private lateinit var iterator: Iterator<PartiQLValue>
+    private var index: Long = 0
+
+    override fun open() {
+        val r = expr.eval(Record.empty)
+        index = 0
+        iterator = when (r) {
+            is CollectionValue<*> -> r.elements!!.iterator()
+            else -> iterator { yield(r) }
+        }
+    }
+
+    override fun next(): Record? {
+        if (!iterator.hasNext()) {
+            return null
+        }
+        val i = BigInteger.valueOf(index)
+        val v = iterator.next()
+        index += 1
+        return Record.of(v, intValue(i))
```

Comment:
From what we discussed, I thought we would be returning an in64Value

Reply:
Yes, had to merge that typing udpate to main first